### PR TITLE
Primitive long list

### PR DESCRIPTION
**Pull request by MishaDemianenko**
Introduce primitive long list and use it in places where
List\<Long\> was used before.

### CODE REVIEW

File: community/primitive-collections/src/main/java/org/neo4j/collection/primitive/PrimitiveLongList.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+import java.util.Arrays;
+
+/**
+ * List implementation that holds primitive longs in array that grows on demand.
+ */
+public class PrimitiveLongList implements PrimitiveLongCollection
+{
+    private static final int DEFAULT_SIZE = 8;
+
+    private long[] elements = PrimitiveLongCollections.EMPTY_LONG_ARRAY;
```
Comment (tinwelint) [inline community/primitive-collections/src/main/java/org/neo4j/collection/primitive/PrimitiveLongList.java]: Unnecessary assignment, no?

File: community/primitive-collections/src/main/java/org/neo4j/collection/primitive/PrimitiveLongList.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    {
+        int newCapacity = elements.length << 1;
+        if ( newCapacity < 0 )
+        {
+            throw new IllegalStateException( "Fail to increase list capacity." );
+        }
+        elements = Arrays.copyOf( elements, newCapacity );
+    }
+
+    private class PrimitiveLongListIterator implements PrimitiveLongIterator
```
Comment (tinwelint) [inline community/primitive-collections/src/main/java/org/neo4j/collection/primitive/PrimitiveLongList.java]: Doesn't quite honor the contract. Try instead doing:

```
private class PrimitiveLongListIterator extends PrimitiveLongCollections.PrimitiveLongBaseIterator`
{
    private int cursor;

    @Override
    boolean fetchNext()
    {
        return cursor < size ? next( elements[cursor++] ) : false;
    }
}
```

**Review by tinwelint — CHANGES_REQUESTED** [review]
Nice, some small comments

**Comment by MishaDemianenko** [comment]
@tinwelint updated

**Review by tinwelint — APPROVED** [review]

**Pull request merged**